### PR TITLE
fix: c2 upload fixes (FPLA-1219)

### PR DESCRIPTION
**Pull request by pd-dudkiewicz**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/FPLA-1219

### Change description ###

- added missing permissions
- added missing pba number validation
- fixed history tab labels and application type description typo

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: ccd-definition/AuthorisationCaseField/cafcass.json [hunk 1]
```diff
@@ -294,6 +294,13 @@
     "UserRole": "caseworker-publiclaw-cafcass",
     "CRUD": "CRU"
   },
+  {
+    "LiveFrom": "01/01/2017",
+    "CaseTypeID": "CARE_SUPERVISION_EPO",
+    "CaseFieldID": "c2ApplicationType",
+    "UserRole": "caseworker-publiclaw-cafcass",
+    "CRUD": "CRU"
+  },
```
Comment (swalker125) [inline ccd-definition/AuthorisationCaseField/cafcass.json]: could you just explain these extra permissions to me?

Comment (pd-dudkiewicz) [reply ccd-definition/AuthorisationCaseField/cafcass.json]: this is new part of c2 upload journey, and permissions were missing for some users that are allowed to upload c2

File: ccd-definition/AuthorisationCaseField/cafcass.json [hunk 2]
```diff
@@ -294,6 +294,13 @@
     "UserRole": "caseworker-publiclaw-cafcass",
     "CRUD": "CRU"
   },
+  {
+    "LiveFrom": "01/01/2017",
+    "CaseTypeID": "CARE_SUPERVISION_EPO",
+    "CaseFieldID": "c2ApplicationType",
+    "UserRole": "caseworker-publiclaw-cafcass",
+    "CRUD": "CRU"
```
Comment (lewisbirks) [inline ccd-definition/AuthorisationCaseField/cafcass.json]: Are they going to need the `D` 😉

File: ccd-definition/AuthorisationCaseField/cafcass.json (hunk 1 above)
Comment (swalker125) [reply ccd-definition/AuthorisationCaseField/cafcass.json]: ok, can we just check for cafcass? I think I remember being told cafcass weren't going to be on the system when we were doing orders and directions

File: ccd-definition/AuthorisationCaseField/court-admin.json [hunk 3]
```diff
@@ -356,6 +356,13 @@
     "UserRole": "caseworker-publiclaw-courtadmin",
     "CRUD": "CRU"
   },
+  {
```
Comment (swalker125) [inline ccd-definition/AuthorisationCaseField/court-admin.json]: small, but this diff isn't necessary if we move it back to where it was originally

File: ccd-definition/AuthorisationCaseField/cafcass.json (hunk 2 above)
Comment (pd-dudkiewicz) [reply ccd-definition/AuthorisationCaseField/cafcass.json]: copy-pasted existing lasolicitor permissions, but after checking it looks like it's unnecessary, so removing

File: ccd-definition/AuthorisationCaseField/court-admin.json (hunk 3 above)
Comment (pd-dudkiewicz) [reply ccd-definition/AuthorisationCaseField/court-admin.json]: dunno how that happened 😄 reverting

File: ccd-definition/AuthorisationCaseField/cafcass.json (hunk 1 above)
Comment (pd-dudkiewicz) [reply ccd-definition/AuthorisationCaseField/cafcass.json]: There are existing cafcass permissions for `c2DocumentBundle` and `temporaryC2Document` which are parts of only C2 upload journey, so I assume this is also needed. Even if cafcass shouldn't have access to C2 upload, it's still not part of this PR.

File: ccd-definition/AuthorisationCaseField/la-solicitor.json [hunk 4]
```diff
@@ -778,7 +785,7 @@
     "CaseTypeID": "CARE_SUPERVISION_EPO",
     "CaseFieldID": "amountToPay",
     "UserRole": "[LASOLICITOR]",
-    "CRUD": "CRUD"
+    "CRUD": "CRU"
```
Comment (lewisbirks) [inline ccd-definition/AuthorisationCaseField/la-solicitor.json]: Why remove the `D` 😉 😕 
Aren't you going to want to remove the field?

Comment (pd-dudkiewicz) [reply ccd-definition/AuthorisationCaseField/la-solicitor.json]: Checked case submission and looks like it's not necessary when we populate field in about-to-start event and remove it in about-to-submit

**Pull request merged**